contracts/external/gmxV2/GmxV2IsolationModeTokenVaultV1.sol

```diff
@@ -0,0 +1,421 @@
+        uint256[] calldata _marketIdsPath,
+        uint256 _inputAmountWei,
+        uint256 _minOutputAmountWei,
+        IGenericTraderProxyV1.TraderParam[] memory _tradersPath,
+        IDolomiteMargin.AccountInfo[] memory _makerAccounts,
+        IGenericTraderProxyV1.UserConfig memory _userConfig
+    ) internal override requireNotFrozen() {
+        super._addCollateralAndSwapExactInputForOutput(
+            _fromAccountNumber,
+            _borrowAccountNumber,
+            _marketIdsPath,
+            _inputAmountWei,
+            _minOutputAmountWei,
+            _tradersPath,
+            _makerAccounts,
+            _userConfig
+        );
+    }
+
+    function _swapExactInputForOutputAndRemoveCollateral(
+        uint256 _toAccountNumber,
+        uint256 _borrowAccountNumber,
+        uint256[] calldata _marketIdsPath,
+        uint256 _inputAmountWei,
+        uint256 _minOutputAmountWei,
+        IGenericTraderProxyV1.TraderParam[] memory _tradersPath,
+        IDolomiteMargin.AccountInfo[] memory _makerAccounts,
+        IGenericTraderProxyV1.UserConfig memory _userConfig
+    )
+    internal override requireNotFrozen() {
+        super._swapExactInputForOutputAndRemoveCollateral(
+            _toAccountNumber,
+            _borrowAccountNumber,
+            _marketIdsPath,
+            _inputAmountWei,
+            _minOutputAmountWei,
+            _tradersPath,
+            _makerAccounts,
+            _userConfig
+        );
+    }
+
+    function _swapExactInputForOutput(
+        uint256 _tradeAccountNumber,
+        uint256[] calldata _marketIdsPath,
+        uint256 _inputAmountWei,
+        uint256 _minOutputAmountWei,
+        IGenericTraderProxyV1.TraderParam[] memory _tradersPath,
+        IDolomiteMargin.AccountInfo[] memory _makerAccounts,
+        IGenericTraderProxyV1.UserConfig memory _userConfig
+    )
+    internal override requireNotFrozen() {
+        super._swapExactInputForOutput(
+            _tradeAccountNumber,
+            _marketIdsPath,
+            _inputAmountWei,
+            _minOutputAmountWei,
+            _tradersPath,
+            _makerAccounts,
+            _userConfig
+        );
+    }
+
+    function _setVirtualBalance(uint256 _bal) internal {
+        _setUint256(_VIRTUAL_BALANCE_SLOT, _bal);
+    }
+
+    function _setVaultFrozen(bool _vaultFrozen) internal {
+        _setUint256(_VAULT_FROZEN_SLOT, _vaultFrozen ? 1 : 0);
+    }
+
+    function _setSourceIsWrapper(bool _sourceIsWrapper) internal {
+        _setUint256(_SOURCE_IS_WRAPPER_SLOT, _sourceIsWrapper ? 1 : 0);
+    }
+
+    function _setShouldSkipTransfer(bool _shouldSkipTransfer) internal {
+        _setUint256(_SHOULD_SKIP_TRANSFER_SLOT, _shouldSkipTransfer ? 1 : 0);
+    }
+
+    function _compareVirtualToRealBalance() internal view {
+        Require.that(
+            getVirtualBalance() == IERC20(UNDERLYING_TOKEN()).balanceOf(address(this)),
+            _FILE,
+            "Virtual and real balance error"
```

Comment:
--> `"Virtual vs real balance mismatch"` (looking for a better description that fits into 32 bytes